src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -0,0 +1,138 @@
+(function() {
+  goog.provide('ga_backgroundselector_directive');
+
+  goog.require('ga_map');
+  goog.require('ga_permalink');
+
+  var module = angular.module('ga_backgroundselector_directive', [
+    'ga_map',
+    'ga_permalink',
+    'pascalprecht.translate'
+  ]);
+
+  module.directive('gaBackgroundSelector',
+    function($document, gaPermalink, gaLayers, gaLayerFilters,
+      gaBrowserSniffer) {
+      return {
+        restrict: 'A',
+        templateUrl:
+            'components/backgroundselector/partials/backgroundselector.html',
+        scope: {
+          map: '=gaBackgroundSelectorMap'
+        },
+        link: function(scope, elt, attrs) {
+          scope.isBackgroundSelectorClosed = true;
+          scope.embed = gaBrowserSniffer.embed;
+          var map = scope.map;
+          var firstLayerChangeEvent = true;
```

Comment:
Instead of this guard variable, you could use `scope.currentTopic` variable...and this currentTopic variable should be function variable and not attached to scope.